torch/_library/autograd.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any, Callable
+
+from .. import _C, autograd
+
+
+def make_autograd_impl(opdef: Any) -> Callable:
+    def autograd_impl(*args):
+        name: str = "GeneratedBackwardFor" + opdef._name
+
+        def forward(ctx, *args):
+            with _C._AutoDispatchBelowAutograd():
+                result = opdef._opoverload(*args)
```

Comment:
This should really be a redispatch...

Reply:
Hmm let me see how difficult it is to add a redispatch to the Python api

Reply:
This seems like it is going to be difficult. Also, what exactly does this change?

I don't think that changing this from a dispatch to a redispatch later will be too BC-breaking (because we can't register an implementation in Python that redispatches before/after and users generally don't know what a redispatch is).

Reply:
Alban:
- how does this interact with backendselect? (there are some subtle interactions).
- how does it interact with things "around the dispatcher", aka the profiler. You'll get two events in the profiler!
- also, the TLSSnapshot thing is unclear(?) but probably OK.

Alban: OK with shipping this first as long as we do the redispatch next, before this thing goes fully public. Blocker for wider adoption (TODO(rzou): file an issue)